### ISSUE TITLE
Command line option `-?` considered as wildcard

### ISSUE THREAD

**Issue by ctron** [fish-shell/fish-shell#7167]
```
$ fish --version
fish, version 3.1.2
$ uname -a
Linux brocken 5.6.19-300.fc32.x86_64 #1 SMP Wed Jun 17 16:10:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

Using `-?` as a command line argument, which is common for getting help from command line tools, fails when using `fish`, as it tries to evaluate it as a wildcard (also with plain fish, no extensions loaded):

~~~
$ git -?
fish: No matches for wildcard “-?”. See `help expand`.
git -?
    ^
~~~

This is not the case in "sh" and "bash". This is an issue with "zsh" as well, but there you have the ability to disable this behavior with an option: https://github.com/ohmyzsh/ohmyzsh/issues/31#issuecomment-359728582

**Comment by faho** [comment]
Duplicate of #4520

**Comment by faho** [comment]
TL;DR: It's deprecated and can be disabled via the "qmark-noglob" feature flag (`set -U fish_features qmark-noglob`). That flag will default to on and then be removed in the future.

**Comment by faho** [comment]
Also important to note that bash *does* try to evaluate the wildcard, only if it doesn't match it passes the unexpanded value along.

Create a file called e.g. `-x` and be amazed.
